### ISSUE TITLE
[remux] remux does not survive reboot on RM2 under 2.9 + toltec testing

### ISSUE THREAD

**Issue by qdatk** [rmkit-dev/rmkit#141]
Installed with the testing branch of Toltec with rm2fb 1:0.0.6-1. Everything works as expected the first time the RM2 starts after using the `systemctl enable remux --now` command, but after subsequent reboots, the swipe gesture no longer works.

- Running `systemctl enable remux --now` does not fix it.
- Running `rm2fb-client remux` brings up the launcher but breaks Xochitl (no longer responsive and has to be rebooted).
- `systemctl status remux` throws up a few instances of this error: `xochitl.gestures Too many touch groups`.

I can fix this by removing and installing remux again via Toltec.

**Comment by Pemo11235** [comment]
I have the same issue. I need to start  the `remux` or `genie` service with `systemctl enable` 
to actually let them work.

**Comment by raisjn** [comment]
can you share the logs after a reboot? `journalctl -u remux`

it's possible that there's an error and it gets disabled? or its possible that the boot process now removes services.

**Comment by matteodelabre** [comment]
(On rM2, 2.9.1)
<details>
<summary>Logs after reboot</summary>

<pre>
-- Reboot --
Aug 29 21:55:52 systemd[1]: Started App launcher that supports multi-tasking applications.
Aug 29 21:55:52 remux[196]: /dev/input/event2 IS UNKNOWN EVENT DEVICE
Aug 29 21:55:52 remux[196]: READING CONFIG FROM /home/root/.config/remux/remux.conf
Aug 29 21:55:53 remux[196]: No such file or directory
Aug 29 21:55:53 remux[196]: No such file or directory
Aug 29 21:55:53 remux[196]: CURRENT APP IS Remarkable
Aug 29 21:55:53 remux[196]: READING TIMEOUTS FOR SLEEP/SHUTDOWN FROM XOCHITL
Aug 29 21:55:53 remux[196]: STARTING SUSPEND THREAD
Aug 29 21:55:53 remux[196]: STARTING FIFO THREAD
Aug 29 21:55:53 remux[196]: LAST SUSPEND LOOP WAS 1630274153 AGO, RESETTING IDLE TIMER
Aug 29 21:55:53 remux[196]: SETTING UP GESTURES
Aug 29 21:55:53 remux[196]: READING CONFIG FROM /home/root/.config/remux/remux.conf
Aug 29 21:55:53 remux[196]: ADDED TAP GESTURE:
Aug 29 21:55:53 remux[196]:   command:
Aug 29 21:55:53 remux[196]:   gesture: tap
Aug 29 21:55:53 remux[196]:   fingers: 3
Aug 29 21:55:53 remux[196]:   zone: -1 -1 -1 -1
Aug 29 21:55:53 remux[196]:   duration: 0
Aug 29 21:55:53 remux[196]: ADDED TAP GESTURE:
Aug 29 21:55:53 remux[196]:   command:
Aug 29 21:55:53 remux[196]:   gesture: tap
Aug 29 21:55:53 remux[196]:   fingers: 4
Aug 29 21:55:53 remux[196]:   zone: -1 -1 -1 -1
Aug 29 21:55:53 remux[196]:   duration: 0
</pre>
</details>

<details>
<summary>Logs after <tt>systemctl restart remux</tt></summary>

<pre>
Aug 29 21:56:54 systemd[1]: Started App launcher that supports multi-tasking applications.
Aug 29 21:56:54 remux[292]: READING CONFIG FROM /home/root/.config/remux/remux.conf
Aug 29 21:56:54 remux[292]: No such file or directory
Aug 29 21:56:54 remux[292]: No such file or directory
Aug 29 21:56:54 remux[292]: READING TIMEOUTS FOR SLEEP/SHUTDOWN FROM XOCHITL
Aug 29 21:56:54 remux[292]: STARTING SUSPEND THREAD
Aug 29 21:56:54 remux[292]: STARTING FIFO THREAD
Aug 29 21:56:54 remux[292]: LAST SUSPEND LOOP WAS 1630274214 AGO, RESETTING IDLE TIMER
Aug 29 21:56:54 remux[292]: SETTING UP GESTURES
Aug 29 21:56:54 remux[292]: READING CONFIG FROM /home/root/.config/remux/remux.conf
Aug 29 21:56:54 remux[292]: ADDED TAP GESTURE:
Aug 29 21:56:54 remux[292]:   command:
Aug 29 21:56:54 remux[292]:   gesture: tap
Aug 29 21:56:54 remux[292]:   fingers: 3
Aug 29 21:56:54 remux[292]:   zone: -1 -1 -1 -1
Aug 29 21:56:54 remux[292]:   duration: 0
Aug 29 21:56:54 remux[292]: ADDED TAP GESTURE:
Aug 29 21:56:54 remux[292]:   command:
Aug 29 21:56:54 remux[292]:   gesture: tap
Aug 29 21:56:54 remux[292]:   fingers: 4
Aug 29 21:56:54 remux[292]:   zone: -1 -1 -1 -1
Aug 29 21:56:54 remux[292]:   duration: 0
</pre>
</details>

In the after-reboot logs, the following line is probably of interest:

```
/dev/input/event2 IS UNKNOWN EVENT DEVICE
```

**Comment by matteodelabre** [comment]
This appears to be caused by the remux service starting too soon at boot time, before the `/dev/input/event2` device is even available. To fix this, we can make the service depend on [`.device` units](https://www.freedesktop.org/software/systemd/man/systemd.device.html). First, we need to make systemd track input devices as `.device` units by adding an udev rule. Create a file called `/lib/udev/rules.d/15-systemd-input.rules` with the following contents:

```udev
ACTION=="add", SUBSYSTEM=="input", TAG+="systemd"
```

Then we can add the required dependencies in remux’s service definition. Edit the file at `/lib/systemd/system/remux.service` as follows:

```diff
 [Unit]
 Description=App launcher that supports multi-tasking applications
-After=xochitl.service opt.mount
+Requires=dev-input-event0.device dev-input-event1.device dev-input-event2.device
+After=xochitl.service opt.mount dev-input-event0.device dev-input-event1.device dev-input-event2.device
 StartLimitInterval=30
 StartLimitBurst=5
 Conflicts=draft.service tarnish.service
```

Then reboot. This fixes the issue, at least on my device. Can any of you try this to see if it fixes your issue? @qdatk @Pemo11235

**Comment by Pemo11235** [comment]
It works for `remux` but not for `genie` (of course).
Is `genie` need same edit?

**Comment by matteodelabre** [comment]
Great! Thanks for testing this. I haven’t tried it with genie yet, but I guess adding the same `dev-input-event*.device` dependencies to the `Requires=` and `After=` fields of its service unit should work too.

**Comment by bestockconsulting** [comment]
> This appears to be caused by the remux service starting too soon at boot time, before the `/dev/input/event2` device is even available. To fix this, we can make the service depend on [`.device` units](https://www.freedesktop.org/software/systemd/man/systemd.device.html). First, we need to make systemd track input devices as `.device` units by adding an udev rule. Create a file called `/lib/udev/rules.d/15-systemd-input.rules` with the following contents:
> 
> ```
> ACTION=="add", SUBSYSTEM=="input", TAG+="systemd"
> ```
> 
> Then we can add the required dependencies in remux’s service definition. Edit the file at `/lib/systemd/system/remux.service` as follows:
> 
> ```diff
>  [Unit]
>  Description=App launcher that supports multi-tasking applications
> -After=xochitl.service opt.mount
> +Requires=dev-input-event0.device dev-input-event1.device dev-input-event2.device
> +After=xochitl.service opt.mount dev-input-event0.device dev-input-event1.device dev-input-event2.device
>  StartLimitInterval=30
>  StartLimitBurst=5
>  Conflicts=draft.service tarnish.service
> ```
> 
> Then reboot. This fixes the issue, at least on my device. Can any of you try this to see if it fixes your issue? @qdatk @Pemo11235

Just dropping another note to say this fixed the issue I had with launch remux after a reboot on version 2.9.1.217 if it helps anyone.

**Comment by qdatk** [comment]
> This appears to be caused by the remux service starting too soon at boot time, before the `/dev/input/event2` device is even available. To fix this, we can make the service depend on [`.device` units](https://www.freedesktop.org/software/systemd/man/systemd.device.html). First, we need to make systemd track input devices as `.device` units by adding an udev rule. Create a file called `/lib/udev/rules.d/15-systemd-input.rules` with the following contents:
> 
> ```
> ACTION=="add", SUBSYSTEM=="input", TAG+="systemd"
> ```
> 
> Then we can add the required dependencies in remux’s service definition. Edit the file at `/lib/systemd/system/remux.service` as follows:
> 
> ```diff
>  [Unit]
>  Description=App launcher that supports multi-tasking applications
> -After=xochitl.service opt.mount
> +Requires=dev-input-event0.device dev-input-event1.device dev-input-event2.device
> +After=xochitl.service opt.mount dev-input-event0.device dev-input-event1.device dev-input-event2.device
>  StartLimitInterval=30
>  StartLimitBurst=5
>  Conflicts=draft.service tarnish.service
> ```
> 
> Then reboot. This fixes the issue, at least on my device. Can any of you try this to see if it fixes your issue? @qdatk @Pemo11235

Can confirm working on my end too (2.9.0.210)! Thanks!

**Comment by alicanerdogan** [comment]
I can also confirm that has also fixed on my device (2.9.1.217)
Huge props 👏

**Comment by raisjn** [comment]
let's close out since i think its working for most people
